### PR TITLE
Productionise action rule status

### PR DESCRIPTION
**Pull request by kacperpONS**
# Has the DDL schema changed?
*Pull requests which include schema changes should be labelled with the `schema change` label for visibility*

*The [common entity model pom.xml](ssdc-rm-common-entity-model/pom.xml) file `project.version` must be bumped
appropriately if there are any common entity code changes*

* [ ] This PR includes DDL schema changes and has been labelled correctly, bumping to new schema
  version: 
* [ ] The POM has been updated with an appropriate version bump if required

# Motivation and Context

We need a new column to track the action rule status

# What has changed

- Add action rule status to common entities
- Add patch and rollback scripts
- Regenerate DDL

# How to test?

Run `make dev-build` on this branch, then test the other linked PRs on the ticket

# Links

https://jira.ons.gov.uk/browse/SDCSRM-520

### CODE REVIEW

File: patches/400_add_action_rule_status.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+-- Purpose: Add a status for the Action Rules and prepopulate for existing ARs
+-- Author: Kacper Prywata, Adam Hawtin
+-- ****************************************************************************
+
+ALTER TABLE casev3.action_rule ADD action_rule_status VARCHAR(255)
+CHECK (action_rule_status in ('SCHEDULED','SELECTING_CASES','PROCESSING_CASES','COMPLETED','ERRORED'));
+
+ALTER casev3.action_rule 
+SET status = "SCHEDULED"
+WHERE hasTriggered = 'false';
```
Comment (dbanks91) [inline patches/400_add_action_rule_status.sql:14]:
```suggestion
update casev3.action_rule 
SET action_rule_status = 'SCHEDULED' 
WHERE has_Triggered = 'false';
```

File: patches/400_add_action_rule_status.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+ALTER casev3.action_rule 
+SET status = "COMPLETED"
+WHERE hasTriggered = 'true';
```
Comment (dbanks91) [inline patches/400_add_action_rule_status.sql:18]:
```suggestion
UPDATE casev3.action_rule 
SET action_rule_status = 'COMPLETED'
WHERE has_Triggered = 'true';
```

**Review by dbanks91 — CHANGES_REQUESTED** [review]

File: patches/400_add_action_rule_status.sql (hunk 2 above)
Comment (dbanks91) [reply patches/400_add_action_rule_status.sql:18]: Just had a thought do we need this if we're allowing action rule status to be nullable for now?

Comment (AdamHawtin) [reply patches/400_add_action_rule_status.sql]: We need to do it at some point, doing it in this step means we should be able to immediately follow up with an update to make it non-nullable. 

We need to be careful about using these scripts to do data migrations, but I believe this example is safe to do as it is on a table with a limited number of rows, and is a non-destructive change, only populating values in a new column.

**Review by ryangrundy7 — APPROVED** [review]

**Review by dbanks91 — APPROVED** [review]

**Pull request merged**